### ISSUE TITLE
Kiwix-serve random feature returns partly broken URLs

### ISSUE THREAD

**Issue by kelson42** [kiwix/kiwix-tools#307]
With a ZIM with the article `?_(album_de_Nena)` from WPFR. The random button returns the following URL: http://localhost:8080/wikipedia_fr_articles_2019-07/A/?_(album_de_Nena) and it should return http://localhost:8080/wikipedia_fr_articles_2019-07/A/%3F_(album_de_Nena).

**Comment by kelson42** [comment]
@mgautierfr The problem here is quite fundamental and is related to URL escaping function, but these functions are not in cause here. The problem is that you can not apply to the whole path escaping properly, you need to apply to each part (namespace, zim url) differently... at the kiwix-serve/kiwix-lib level, you don't have this information anymore!

**Comment by kelson42** [comment]
@mgautierfr Moving the random feature in the libzim should help fixing this bug properly right?

**Comment by mgautierfr** [comment]
No, the random feature in libzim will just keep a random entry in the zim file.
The generation of the url will still be in kiwix-lib side.

(I'm moving the issue in kiwix-lib repository)
